[seanngja] iP

seanngja opened the pull request.

seanngja committed:
Rename Duke to Dobby, Implement basic function

seanngja committed:
Remove Duke traces

seanngja committed:
Add echo functionality and exit command to Dobby

seanngja committed:
Add ability to store and display user text

seanngja committed:
Add ability to mark tasks as done

- Introduced the Task class to represent individual tasks with a
  description and completion status
- Updated Dobby class to handle adding tasks, listing tasks and marking
  tasks as done

seanngja committed:
Update coding standards, add ability to unmark tasks

seanngja committed:
Add support for tracking ToDos, Deadlines and Events

1. ToDos: tasks without any date/time attached
2. Deadlines: tasks that need to be done before a specific timing
3. Events: tasks that start and end at specific times

seanngja committed:
Add Automated Text UI Testing

seanngja committed:
Improve code quality

Refactor code in main function to improve readability

shaunngoh commented on src/main/java/Dobby.java:
```diff
@@ -0,0 +1,138 @@
+
+public class Dobby {
+
+    private static final String DASH_LINE = "____________________________________________________________";
```
I like that you followed the convention by naming a constant in all caps

shaunngoh commented on src/main/java/Dobby.java:
```diff
@@ -0,0 +1,138 @@
+        }
+    }
+
+    private static void printWelcomeMessage(){
```
There is a missing space for your method.
```suggestion
    private static void printWelcomeMessage() {
```

shaunngoh commented on src/main/java/Dobby.java:
```diff
@@ -0,0 +1,138 @@
+        printSeparator();
+    }
+
+    private static void addTask(String line){
```
There is a missing space for the method. 
```suggestion
    private static void addTask(String line) {
```

shaunngoh commented on src/main/java/Dobby.java:
```diff
@@ -0,0 +1,138 @@
+        printTaskAddedMessage();
+    }
+
+    private static Task createTask(String line){
```
There is a missing white space for the method
```suggestion
    private static Task createTask(String line) {
```

shaunngoh commented on src/main/java/Dobby.java:
```diff
@@ -0,0 +1,138 @@
+        return new Task(line);
+    }
+
+    private static Deadline createDeadlineTask(String line){
```
There is a missing white space for the method
```suggestion
    private static Deadline createDeadlineTask(String line) {
```

shaunngoh reviewed (commented):
Overall, goodjob! You followed the naming and coding conventions. I also found that your code was easy to read and to follow.

anlamm commented on src/main/java/Dobby.java:
```diff
@@ -0,0 +1,138 @@
+            return createDeadlineTask(line);
+        } else if (line.startsWith("event ")){
+            return createEventTask(line);
+        }
```
Maybe better to put an "else" here?

anlamm commented on src/main/java/Dobby.java:
```diff
@@ -0,0 +1,138 @@
+
+    private static int listSize = 0;
+
+    public static void main(String[] args) {
```
Methods are of suitable length. The abstraction is also clear and understandable

anlamm commented on src/main/java/Dobby.java:
```diff
@@ -0,0 +1,138 @@
+        String line;
+
+        printWelcomeMessage();
+
```
I like that you add lines between printWelcomeMessage() and printGoodbyeMessage() to make the outline more clear

anlamm commented on src/main/java/Dobby.java:
```diff
@@ -0,0 +1,138 @@
+
+    private static void unmarkTaskAsDone(String line){
+        int taskNumber = Integer.parseInt(line.substring(line.lastIndexOf(" ") + 1));
+        if (taskNumber > 0 && taskNumber <= listSize){
```
maybe better to also handle the case when taskNumber is out of range

seanngja committed:
Update coding standard

Added spaces between ending bracket of function definition ')' and
opening curly brace '{' of function

seanngja committed:
Add ability to handle errors

1) Update code in main to handle exceptions for missing description,
illegal inputs, marking marked tasks and unmarking unmarked task.
2) Added a getter in Task object as well

seanngja committed:
Update error handling for full and empty task list

seanngja committed:
Merge branch 'branch-Level-5' into branch 'master'

Merged updates in branch back to master without a fast-forward

seanngja committed:
Divide classes into packages

- Organize classes into 'tasks' and 'command' packages.
- Update 'runtest.bat' to compile and run correctly.

seanngja committed:
Merge branch 'branch-A-Packages' into 'master'

seanngja committed:
Updated taskList from array to ArrayList

seanngja committed:
Add support for deleting tasks from the list

seanngja committed:
Add ability to save and load tasks from hard disk

- taskList now remembers tasks added from previous runs
- Update gitignore to not save /data/ folder
- Add getters for fromTime and toTime for Events

seanngja committed:
Merge branch 'branch-Level-6'

Add ability to delete tasks

seanngja committed:
Merge branch 'branch-Level-7'

seanngja committed:
Update commands to save after each command

- Task list updates at end of each command instead of only after bye
is called

Kobot7 commented on src/main/java/dobby/command/Dobby.java:
```diff
@@ -0,0 +1,316 @@
+                }
+                try {
+                    processCommand(line);
+                } catch (Exception e) {
```
Is the variable name "e" informative enough?

Kobot7 commented on src/main/java/dobby/command/Dobby.java:
```diff
@@ -0,0 +1,316 @@
+
+    private static void processCommand (String line)
+            throws MissingDescriptionException, IllegalInputException, TaskAlreadyMarkedException,
+                TaskAlreadyUnmarkedException, EmptyListException {
```
Indentation for wrapped lines should be 8 spaces more than the parent line (line 146)

Kobot7 commented on src/main/java/dobby/command/Dobby.java:
```diff
@@ -0,0 +1,316 @@
+
+    }
+
+    private static void loadTasks() {
+        File directory = new File(DIRECTORY_PATH);
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+
+        File file = new File(FILE_PATH);
+        if (file.exists()) {
+            try (Scanner scanner = new Scanner(file)) {
+                while (scanner.hasNextLine()) {
+                    String line = scanner.nextLine();
+                    Task task = parseTaskFromFile(line);
+                    if (task != null) {
+                        taskList.add(task);
+                    }
+                }
+            } catch (FileNotFoundException e) {
+                System.out.println("File not found");
+            }
+        }
+    }
```
Try to avoid deep nesting

Kobot7 commented on src/main/java/dobby/command/Dobby.java:
```diff
@@ -0,0 +1,316 @@
+    }
+
+    private static Task parseTaskFromFile(String line) {
+        String[] parts = line.split("\\|");
```
Consider using a named constant instead of a magic strings

seanngja committed:
Add Ui, Storage, Parser, TaskList and TaskCreator classes

Code refactoring to extract out closely related code as classes

seanngja committed:
Improve code quality

seanngja committed:
Add ability to process dates and datetimes

seanngja committed:
Add ability to find tasks by searching keyword

- format: find <keyword>
- list out all tasks containing <keyword>

seanngja committed:
Add JavaDoc comments to the code

seanngja committed:
Merge pull request #1 from seanngja/branch-Level-8

Add ability to process dates and datetimes

seanngja committed:
Merge pull request #2 from seanngja/branch-Level-9

Add ability to find tasks by searching keyword

seanngja committed:
Merge branch 'master' into branch-A-JavaDoc

seanngja committed:
Merge pull request #3 from seanngja/branch-A-JavaDoc

Add JavaDoc comments to the code

seanngja committed:
Add a basic user guide

seanngja committed:
Update user guide

seanngja committed:
Update handling of more invalid inputs

- Update error handling when task number inputted is invalid for
  marking, unmarking and deleting tasks
- Strip excess white spaces when creating and storing tasks

seanngja committed:
Move delete task print message to Ui object

seanngja committed:
Update .gitignore to ignore /text-ui-test/data/
